yt_dlp/extractor/bilibili.py

```diff
@@ -152,11 +178,50 @@ def _get_episodes_from_season(self, ss_id, url):
             note='Downloading season info', query={'season_id': ss_id},
             headers={'Referer': url, **self.geo_verification_headers()})
 
-        for entry in traverse_obj(season_info, (
-                'result', 'main_section', 'episodes',
-                lambda _, v: url_or_none(v['share_url']) and v['id'])):
+        for entry in traverse_obj(season_info, ('result', 'main_section', 'episodes',
+                                                lambda _, v: url_or_none(v['share_url']) and v['id'])):
             yield self.url_result(entry['share_url'], BiliBiliBangumiIE, f'ep{entry["id"]}')
 
+    def _enc_wbi(self, params: dict, video_id=None):
+        if video_id is None:
+            video_id = 0
+        session_data = self._download_json('https://api.bilibili.com/x/web-interface/nav',
+                                           video_id, note='wbi signature...', fatal=False)
+
+        key_from_url = lambda x: x[x.rfind('/') + 1:].split('.')[0]
+        img_key = traverse_obj(
+            session_data, ('data', 'wbi_img', 'img_url', {key_from_url}))
+        sub_key = traverse_obj(
+            session_data, ('data', 'wbi_img', 'sub_url', {key_from_url}))
```

Comment:
I think it makes sense to concatenate them in traversal:
```suggestion
        lookup = traverse_obj(
            session_data, ('data', 'wbi_img', ('img_url', 'sub_url'), {key_from_url}, ...))
```
This also means we could inline the conversion instead as well:
```py
        lookup = traverse_obj(session_data, (
            'data', 'wbi_img', ('img_url', 'sub_url'),
            {lambda x: x.rpartition('/')[2].partition('.')[0]}, ...))
```

Reply:
`wbi_img` is a dict, so I delete `...` and it works again.

Reply:
Sorry, mixup again. What I wanted to do was
```py
        lookup = ''.join(traverse_obj(session_data, (
            'data', 'wbi_img', ('img_url', 'sub_url'),
            {lambda x: x.rpartition('/')[2].partition('.')[0]})))
```
As a helper to try and understand how it works, please look at this:
```py
>>> data = {"x": {"a": "a/test_a", "b": "b/test_b"}}
>>> ''.join(traverse_obj(data, ("x", ("a", "b"), {lambda x: x.partition("/")[2]})))
'test_atest_b'
```
As you can see, it now concatenates them into a single string, which is what you did below using `+`